src/Nested.ts

```diff
@@ -1,5 +1,11 @@
 import {ConstraintDefinitionError, getJoiSchema, Joi, updateSchema} from "./core";
 
+export class NestedPropertyTypeUnknown extends ConstraintDefinitionError {
```

Comment:
I wanted to get your opinion on this.

There were a couple error messages repeated through the code which I decided to turn into their own error classes descending from `ConstraintDefinitionError`. They are `NestedPropertyTypeUnknown` and `ValidationSchemaNotFound`.

I'm not sure but there's a chance these could actually be breaking changes for consumers doing `instanceof` checks for `ConstraintDefinitionError`. Instead I could turn these classes into functions that just return a `new ConstraintDefinitionError` with the appropriate message? Thoughts?

Reply:
Looks good to me. I can't think why anyone would be checking for these errors, since they should only come up if a developer has mis-decorated something, which will get identified and fixed pretty quickly.

Reply:
Okay, in that case I should also ask, should I give these errors their own `name`? Currently they will inherit the `name` `"ConstraintDefinitionError"`.

Reply:
Yes, sounds good.